### DISCUSSION TITLE
Handling the Password Reset for a user by Supabase itself.

### DISCUSSION THREAD

**Discussion by MBSA-INFINITY** [supabase/supabase#15044] [Feature Requests]
**The Problem**
The function `supabase.auth.reset_password_email(email)` in **supabase-py** only sends a reset mail to the particular email address mentioned, but does not actually handles the password reset of that account, unlike firebase. This function `reset_password_email()` should not only just send a password reset mail but also handle password reset with a unique form link specific to that user, so that the particular user can change the password.

The javascript function `.resetPasswordForEmail()`  takes in `redirectTo` parameter and this particular link is present as a password reset link in the email that supabase sends. But instead of this, if supabase handles the password reset itself, this would be a great help for the users because then they would not have to create customer password reset links and handle password reset from their end.

**The solution**
The function `reset_password_email(email)` should generate a unique supabase link which would contain a form to reset the password and this link should be sent to the particular mail as a password reset mail. This way supabase users wouldn't have to worry about password reset themselves. They just have to call this function and rest, we would handle. ***Firebase has the same feature already!***

**Additional context**
Do checkout this Loom:- [Firebase Password Reset Flow](https://www.loom.com/share/211759ff187e4a8f9dd5d14dd434c7e5)

**Comment by Akilesh1989** [comment]
This will be a very useful feature
